src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -31,5 +48,41 @@ public abstract class CompilationModuleGroup
         /// accessed through the target platform's import mechanism (ie, Import Address Table on Windows)
         /// </summary>
         public abstract bool ShouldReferenceThroughImportTable(TypeDesc type);
+
+        private class CompilerGeneratedAssembly : ModuleDesc, IAssemblyDesc
```

Comment:
I think there's a value in keeping this collocated with the place that used this. This is an internal implementation detail (same as e.g. `ExplicitScopeAssembly` or `DynamicInvokeThunkGenericParameter`). It doesn't seem right to make these more accessible than necessary (and place them further away from the place where they're used) just because they derive from a typesystem class. I find nested classes a good way to keep implementation details private.